metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/HiveConnectorFastTableService.java

```diff
@@ -114,43 +89,59 @@ public boolean exists(final ConnectorRequestContext requestContext, final Qualif
         final List<String> uris,
         final boolean prefixSearch
     ) {
-        final long start = registry.clock().wallTime();
-        // Create the sql
-        final StringBuilder queryBuilder = new StringBuilder(SQL_GET_TABLE_NAMES_BY_URI);
-        final List<SqlParameterValue> params = Lists.newArrayList();
-        if (prefixSearch) {
-            queryBuilder.append(" and (1=0");
-            uris.forEach(uri -> {
-                queryBuilder.append(" or location like ?");
-                params.add(new SqlParameterValue(Types.VARCHAR, uri + "%"));
-            });
-            queryBuilder.append(" )");
+        return directSqlTable.getTableNames(uris, prefixSearch);
+    }
+
+    /**
+     * Update a table with the given metadata.
+     *
+     * If table is an iceberg table, then lock the table for update so that no other request can update it. If the meta
+     * information is invalid, then throw an error.
+     * If table is not an iceberg table, then do a regular table update.
+     *
+     * @param requestContext The request context
+     * @param tableInfo      The resource metadata
+     */
+    @Override
+    public void update(final ConnectorRequestContext requestContext, final TableInfo tableInfo) {
+        if (isIcebergTable(tableInfo)) {
+            final QualifiedName tableName = tableInfo.getName();
+            final Long tableId = directSqlTable.getTableId(tableName);
+            try {
+                directSqlTable.lockIcebergTable(tableId);
+                try {
+                    final TableInfo existingTableInfo = get(requestContext, tableInfo.getName());
+                    if (isIcebergTable(existingTableInfo) && isValidIcebergUpdate(existingTableInfo, tableInfo)) {
```

Comment:
Should ` isIcebergTable(existingTableInfo)` be moved to the top? If the existing table not an icebergTable, are we able to lock it? I guess, by inserting a location? But is this valid?  To valid a table to be iceberg should be stay in one function, maybe we can have one function to check both the new table info request and the existing table are both valid iceberg table.